File: basic-to-do/src/main/java/com/projects/basictodo/services/TareaService.java
package com.projects.basictodo.services;

import java.util.List;

import com.projects.basictodo.models.Tarea;

public interface TareaService {

	public List<Tarea> findAllTareas();
	
	public void saveTareas(Tarea tarea);
	
	public void updateDescripcionTarea(Long idTarea, String descripcion);
	
	public void confirmEstadoTarea(Long idTarea, boolean flg_estado);
	
	public void deleteTareaByID(Long idTarea);
	
}

File: Frontend/basictodoapp/src/app/tarea.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Tarea } from './tarea';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})

export class TareaService {
  private apiServerUrl = environment.apiBaseUrl;
  
  constructor(private http: HttpClient) { }
  
  public getListTarea(): Observable<Tarea[]> {
    return this.http.get<Tarea[]>(`${this.apiServerUrl}/rest/findAllTareas`);
  }
  
  public addTarea(tarea: Tarea): Observable<Tarea> {
    return this.http.post<Tarea>(`${this.apiServerUrl}/rest/saveTarea`, tarea);
  }

  public updateDescripcion(tarea: Tarea): Observable<Tarea> {
    return this.http.put<Tarea>(`${this.apiServerUrl}/rest/updateDescripcionTarea`, tarea);
  }

  public confirmTarea(tarea: Tarea): Observable<Tarea> {
    return this.http.put<Tarea>(`${this.apiServerUrl}/rest/confirmEstadoTarea`, tarea);
  }

  public deleteTarea(idTarea: number): Observable<void> {
    return this.http.delete<void>(`${this.apiServerUrl}/rest/deleteTareaByID/${idTarea}`);
  }

}